src/OpenLoco/src/Vehicles/VehicleDraw.cpp

```diff
@@ -403,18 +521,118 @@ namespace OpenLoco
+    static ColourScheme getCompanyVehicleColourScheme(CompanyId companyId, const VehicleObject& vehObject)
     {
-        auto* vehObject = ObjectManager::get<VehicleObject>(vehicleTypeIdx);
         auto* company = CompanyManager::get(companyId);
 
         auto colourScheme = company->mainColours;
-        if (company->customVehicleColoursSet & (1 << vehObject->colourType))
+        if (company->customVehicleColoursSet & (1 << vehObject.colourType))
         {
-            colourScheme = company->vehicleColours[vehObject->colourType - 1];
+            colourScheme = company->vehicleColours[vehObject.colourType - 1];
         }
+        return colourScheme;
+    }
+
+    // 0x004B7741
+    void drawVehicleOverview(Gfx::DrawingContext& drawingCtx, Ui::Point offset, int16_t vehicleTypeIdx, uint8_t yaw, uint8_t roll, CompanyId companyId)
+    {
+        auto* vehObject = ObjectManager::get<VehicleObject>(vehicleTypeIdx);
+
+        auto colourScheme = getCompanyVehicleColourScheme(companyId, *vehObject);
 
         drawVehicleOverview(drawingCtx, offset, *vehObject, yaw, roll, colourScheme);
     }
+
+    static DrawItems toScreenDistDrawItems(const DrawItems& drawItems, const uint8_t yaw)
+    {
+        DrawItems screenDistDrawItems{};
+
+        const auto getScreenDistance = [](int32_t gameDist, uint8_t yaw) {
+            const auto unk1 = Math::Trigonometry::computeXYVector(gameDist, yaw) / 4;
+
+            const auto p1 = World::gameToScreen(World::Pos3(unk1.x, unk1.y, 0), 0);
+            return -p1.x;
+        };
+
+        screenDistDrawItems.totalDistance = getScreenDistance(drawItems.totalDistance, yaw);
+
+        for (auto& item : drawItems.items)
+        {
+            screenDistDrawItems.items.push_back(DrawItem{ item.image, getScreenDistance(item.dist, yaw), item.isBody });
+        }
+        return screenDistDrawItems;
+    }
+
+    // 0x004B7711 & 0x004B745B
+    int16_t drawVehicleInline(Gfx::DrawingContext& drawingCtx, int16_t vehicleTypeIdx, CompanyId company, Ui::Point loc)
+    {
+        // This has been simplified from vanilla.
+
+        loc.y += 19;
+        auto* vehObject = ObjectManager::get<VehicleObject>(vehicleTypeIdx);
+
+        loc.y += getVehicleObjectYDisplayOffset(*vehObject);
+
+        auto colourScheme = getCompanyVehicleColourScheme(company, *vehObject);
+
+        const auto yaw = 40;
+
+        const auto drawItems = getDrawItemsForVehicle(*vehObject, yaw, 0, colourScheme);
+
+        auto screenDistDrawItems = toScreenDistDrawItems(drawItems, yaw);
+
+        for (auto& item : screenDistDrawItems.items)
+        {
+            drawingCtx.drawImage(loc + Ui::Point(item.dist, 0), item.image);
+        }
+        return screenDistDrawItems.totalDistance;
+    }
+
+    // 0x004B6D93
+    int16_t drawVehicleInline(Gfx::DrawingContext* drawingCtx, const Vehicles::Car& car, Ui::Point loc, uint8_t flags, Colour disableColour)
```

Comment:
```suggestion
    int16_t drawVehicleInline(Gfx::DrawingContext& drawingCtx, const Vehicles::Car& car, Ui::Point loc, uint8_t flags, Colour disableColour)
```

Reply:
Its like that on purpose although i think i might redesign it and break out the get width version.